.github/workflows/start-self-hosted-runner/action.yml

```diff
@@ -20,13 +17,24 @@ runs:
         - name: Configure AWS Credentials
           uses: aws-actions/configure-aws-credentials@v4
           with:
-              aws-access-key-id: ${{ inputs.aws-access-key-id }}
-              aws-secret-access-key: ${{ inputs.aws-secret-access-key }}
+              role-to-assume: ${{ inputs.role-to-assume }}
               aws-region: ${{ inputs.aws-region }}
+              
         - name: Start EC2 self hosted runner
           shell: bash
           run: |
               sudo apt update
               sudo apt install awscli -y
-              aws ssm send-command --instance-ids ${{ inputs.ec2-instance-id }} --document-name  StartGithubSelfHostedRunner --output text
-              aws ssm list-command-invocations 
+              command_id=$(aws ssm send-command --instance-ids ${{ inputs.ec2-instance-id }} --document-name StartGithubSelfHostedRunner --query Command.CommandId | tr -d '"')
+            
+              status=""
```

Comment:
$status is an exiting env variable in linux, rename it to invoke_status.
and you don't need to initiate it with empty string, "$invoke_status" will return "" when not set